### ISSUE TITLE
Can not resolve secrets in Datasources

### ISSUE THREAD

**Issue by malakaganga** [wso2/micro-integrator#1601]
**Description:**
Can not resolve secrets in Datasources

**Steps to reproduce:**
Add the following data source in deployment.toml

[[datasource]]
id = "WSO2_COORDINATION_DB"
url= "jdbc:mysql://localhost:3306/clusterdb"
username="root"
password="$secret{db_pass}"
driver="com.mysql.jdbc.Driver"

And when loading the server following exception throws

```Error in initializing Tasks component: Error in connecting to the datasource to create tables for clustering org.wso2.micro.integrator.coordination.exception.ClusterCoordinationException: Error in connecting to the datasource to create tables for clustering
	at org.wso2.micro.integrator.coordination.RDBMSCommunicationBusContextImpl.<init>(RDBMSCommunicationBusContextImpl.java:69)
	at org.wso2.micro.integrator.coordination.RDBMSCoordinationStrategy.<init>(RDBMSCoordinationStrategy.java:134)
	at org.wso2.micro.integrator.coordination.ClusterCoordinator.<init>(ClusterCoordinator.java:40)
	at org.wso2.micro.integrator.ntask.core.internal.TasksDSComponent.activate(TasksDSComponent.java:118)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.eclipse.equinox.internal.ds.model.ServiceComponent.activate(ServiceComponent.java:260)
	at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.activate(ServiceComponentProp.java:146)
	at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.build(ServiceComponentProp.java:345)
	at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponent(InstanceProcess.java:620)
	at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponents(InstanceProcess.java:197)
	at org.eclipse.equinox.internal.ds.Resolver.buildNewlySatisfied(Resolver.java:473)
	at org.eclipse.equinox.internal.ds.Resolver.enableComponents(Resolver.java:217)
	at org.eclipse.equinox.internal.ds.SCRManager.performWork(SCRManager.java:816)
	at org.eclipse.equinox.internal.ds.SCRManager$QueuedJob.dispatch(SCRManager.java:783)
	at org.eclipse.equinox.internal.ds.WorkThread.run(WorkThread.java:89)
	at org.eclipse.equinox.internal.util.impl.tpt.threadpool.Executor.run(Executor.java:70)
Caused by: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)```

**Comment by dulanjalidilmi** [comment]
We should backport this fix for MI 1.1.0 as well. Hence reopening this issue.

**Comment by sachithKay** [comment]
Closing with the fix : https://github.com/wso2/micro-integrator/pull/1611
